optaplanner-docs/src/main/asciidoc/QuickStart/SpringBoot/SpringBootQuickStart-chapter.adoc

```diff
@@ -0,0 +1,1057 @@
+        solverManager.terminateEarly(TimeTableRepository.SINGLETON_TIME_TABLE_ID);
+    }
+
+}
+----
+====
++
+For simplicity's sake, this code handles only one `TimeTable`,
+but it is straightforward to enable multi-tenancy and handle multiple `TimeTable` instances of different high schools in parallel.
++
+The `getTimeTable()` method returns the latest time table from the database.
+It uses the `ScoreManager` (which is automatically injected)
+to calculate the score of that time table, so the UI can show the score.
++
+The `solve()` method starts a job to solve the current time table and store the time slot and room assignments in the database.
+It uses the `SolverManager.solveAndListen()` method to listen to intermediate best solutions
+and update the database accordingly.
+This enables the UI to show progress while the backend is still solving.
+
+. Adjust the `TimeTableControllerTest` accordingly, now that the `solve()` method returns immediately.
+Poll for the latest solution until the solver finishes solving:
++
+====
+[source,java]
+----
+package com.example.solver;
+
+import com.example.domain.Lesson;
+import com.example.domain.TimeTable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.optaplanner.core.api.solver.SolverStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest(properties = {
+        "optaplanner.solver.termination.spent-limit=1h", // Effectively disable this termination in favor of the best-score-limit
+        "optaplanner.solver.termination.best-score-limit=0hard/*soft"})
+public class TimeTableControllerTest {
+
+    @Autowired
+    private TimeTableController timeTableController;
+
+    @Test
+    @Timeout(600_000)
+    public void solveDemoDataUntilFeasible() throws InterruptedException {
+        timeTableController.solve();
+        TimeTable timeTable = timeTableController.getTimeTable();
+        while (timeTable.getSolverStatus() != SolverStatus.NOT_SOLVING) {
+            // Quick polling (not a Test Thread Sleep anti-pattern)
+            // Test is still fast on fast machines and doesn't randomly fail on slow machines.
+            Thread.sleep(20L);
+            timeTable = timeTableController.getTimeTable();
+        }
+        assertFalse(timeTable.getLessonList().isEmpty());
+        for (Lesson lesson : timeTable.getLessonList()) {
+            assertNotNull(lesson.getTimeslot());
+            assertNotNull(lesson.getRoom());
+        }
+        assertTrue(timeTable.getScore().isFeasible());
+    }
+
+}
+----
+====
+
+. Build an attractive web UI on top of these REST methods to visualize the timetable.
+
+Take a look at the example's source code to see how this all turns out.
```

Comment:
I haven't noticed any link for the example's source code. On the other hand, if this documentation ends up in a product doc, we should not refer to any upstream resources there.

Reply:
Yea, but we could refer to the download zip. I 'll adjust that in the next PR.